### PR TITLE
Add test for S3 and Glue metastore

### PR DESCRIPTION
**Pull request by ebyhr**
## Description

Add test for S3 and Glue metastore

## Release notes

(x) This is not user-visible or docs only and no release notes are required.

### CODE REVIEW

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveS3AndGlueMetastoreTest.java [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+    }
+
+    @Override
+    protected void validateFilesAfterOptimize(String location, Set<String> initialFiles, Set<String> updatedFiles)
+    {
+        assertThat(updatedFiles).hasSizeLessThan(initialFiles.size());
+        assertThat(getAllDataFilesFromTableDirectory(location)).isEqualTo(updatedFiles);
+    }
+
+    protected String adaptTableLocation(String location)
```
Comment (findinpath) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveS3AndGlueMetastoreTest.java:118]: Why do we have this method?

**Review by findinpath — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -655,16 +655,18 @@ jobs:
...
           AWS_REGION: us-east-2
+          S3_BUCKET: trino-ci-test
           GCP_CREDENTIALS_KEY: ${{ secrets.GCP_CREDENTIALS_KEY }}
         # Run tests if any of the secrets is present. Do not skip tests when one secret renamed, or secret name has a typo.
         if: >-
           contains(matrix.modules, 'trino-delta-lake') && contains(matrix.profile, 'cloud-tests') &&
           (env.CI_SKIP_SECRETS_PRESENCE_CHECKS != '' || env.ABFS_ACCOUNT != '' || env.ABFS_CONTAINER != '' || env.ABFS_ACCESSKEY != '' || env.AWS_ACCESS_KEY_ID != '' || env.AWS_SECRET_ACCESS_KEY != '' || env.GCP_CREDENTIALS_KEY != '')
         run: |
           $MAVEN test ${MAVEN_TEST} ${{ format('-P {0}', matrix.profile) }} -pl :trino-delta-lake \
+            -Ds3.bucket="${S3_BUCKET}" \
```
Comment (findepi) [inline .github/workflows/ci.yml:669]: or we keep reading it with env variable

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveS3AndGlueMetastoreTest.java [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+
+        assertUpdate("DROP SCHEMA \"" + schemaName + "\"");
+    }
+
+    @Override
+    protected List<String> locationPatterns()
+    {
+        // TODO https://github.com/trinodb/trino/issues/17803 Fix correctness issue with double slash
+        return super.locationPatterns().stream()
+                .filter(location -> !location.contains("double_slash"))
```
Comment (findepi) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveS3AndGlueMetastoreTest.java:295]: This is extracted from https://github.com/trinodb/trino/pull/17804
is this the only place that's different?

**Review by findepi — APPROVED** [review]

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveS3AndGlueMetastoreTest.java (hunk 3 above)
Comment (ebyhr) [reply plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveS3AndGlueMetastoreTest.java:295]: This PR doesn't contain `adaptTableLocation`, `testCreateTableWithDoubleSlash` and `testCtasWithDoubleSlash`.
* `adaptTableLocation` is an extension point when we run the test with HMS in the future. 
* `testCreateTableWithDoubleSlash` and `testCtasWithDoubleSlash` require some changes in `HiveLocationService`

Comment (findepi) [reply plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveS3AndGlueMetastoreTest.java:295]: > * `testCreateTableWithDoubleSlash` and `testCtasWithDoubleSlash` require some changes in `HiveLocationService`

or reverting de-hadooping of HiveLocationService?

**Pull request merged**